processor/transformprocessor/internal/logs/processor.go

```diff
@@ -26,7 +26,7 @@ import (
 )
 
 type Processor struct {
-	queries []ottl.Statement
+	queries []ottl.Statement[ottllogs.TransformContext]
```

Comment:
Maybe we could do something similar here as proposed for `NewParser`?  Have `[]ottllogs.Statements` so that we don't have to have the processor pass in the Generic itself.

Hopefully we can make it so that the only place a processor would need to directly interact with the generic is when creating its own functions.

Reply:
So essentially have a `TracesStatements`?

Reply:
I can also look into this for the next PR when I look into the Parser.

Reply:
Ya something like that.  Since the generic introduce some complexity, will be great to hide that complexity inside the Contexts as much as possible since, at least at the moment, we need a Parser and set of Statements per context.